x-pack/plugins/ml/public/application/explorer/anomaly_detection_alerts_state_service.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BehaviorSubject, combineLatest, type Observable, Subscription } from 'rxjs';
+import { map, startWith, switchMap } from 'rxjs/operators';
+import {
+  DataPublicPluginStart,
+  TimefilterContract,
+  isRunningResponse,
+} from '@kbn/data-plugin/public';
+import type {
+  RuleRegistrySearchRequest,
+  RuleRegistrySearchResponse,
+} from '@kbn/rule-registry-plugin/common';
+import {
+  AlertConsumers,
+  ALERT_RULE_TYPE_ID,
+  ALERT_UUID,
+  ALERT_TIME_RANGE,
+  ALERT_RULE_NAME,
+  ALERT_STATUS,
+} from '@kbn/rule-data-utils';
+import { isDefined } from '@kbn/ml-is-defined';
+import { getSeverityColor } from '@kbn/ml-anomaly-utils';
+import {
+  ALERT_ANOMALY_DETECTION_JOB_ID,
+  ALERT_ANOMALY_SCORE,
+  ALERT_ANOMALY_TIMESTAMP,
+  ML_ALERT_TYPES,
+} from '../../../common/constants/alerts';
+import { StateService } from '../services/state_service';
+import { AnomalyTimelineStateService } from './anomaly_timeline_state_service';
+
+export interface AnomalyDetectionAlert {
+  id: string;
+  anomalyScore: number;
+  jobId: string;
+  anomalyTimestamp: number;
+  timestamp: number;
+  end_timestamp: number;
+  ruleName: string;
+  color: string;
+  alertStatus: string;
+}
+
+export class AnomalyDetectionAlertsStateService extends StateService {
+  private readonly _aadAlerts$ = new BehaviorSubject<AnomalyDetectionAlert[]>([]);
```

Comment:
Should this be `_adAlerts` for anomaly detection alerts? Or what does `aad` signify?

Reply:
It stand for alerts-as-data. Added comments in https://github.com/elastic/kibana/pull/167998/commits/db3c4a527127284301575ae996c0a1ae98d827a6